Saving a project with Bill to customer set to "No customer (reusable project)" crashes

In production saving an existing project, if "No customer (reusable project)" is chosen in the UI crashes:

```
I, [2025-08-05T00:55:16.188846 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb] Started GET "/projects/5" for 100.125.101.25 at 2025-08-05 00:55:16 +0000
I, [2025-08-05T00:55:16.190513 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb] Processing by ProjectsController#show as TURBO_STREAM
I, [2025-08-05T00:55:16.190607 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Parameters: {"id"=>"5"}
D, [2025-08-05T00:55:16.192624 #96471] DEBUG -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Project Load (0.9ms)  SELECT "projects".* FROM "projects" WHERE "projects"."id" = 5 LIMIT 1 /*action='show',application='Abt',controller='projects'*/
D, [2025-08-05T00:55:16.193802 #96471] DEBUG -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Rendering layout layouts/application.html.haml
D, [2025-08-05T00:55:16.193972 #96471] DEBUG -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Rendering projects/show.html.haml within layouts/application
I, [2025-08-05T00:55:16.195155 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Rendered projects/show.html.haml within layouts/application (Duration: 1.1ms | GC: 0.0ms)
I, [2025-08-05T00:55:16.195304 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]   Rendered layout layouts/application.html.haml (Duration: 1.4ms | GC: 0.0ms)
I, [2025-08-05T00:55:16.195559 #96471]  INFO -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb] Completed 500 Internal Server Error in 5ms (ActiveRecord: 0.9ms (1 query, 0 cached) | GC: 0.0ms)
E, [2025-08-05T00:55:16.198089 #96471] ERROR -- : [e46b1d11-2144-4bc5-baa2-630d32338dfb]
[e46b1d11-2144-4bc5-baa2-630d32338dfb] ActionView::Template::Error (undefined method `name' for nil)
[e46b1d11-2144-4bc5-baa2-630d32338dfb] Caused by: NoMethodError (undefined method `name' for nil)
[e46b1d11-2144-4bc5-baa2-630d32338dfb]
Information for: ActionView::Template::Error (undefined method `name' for nil):
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     28:           .col-sm-3
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     29:             %strong Bill to Customer:
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     30:           .col-sm-9
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     31:             = @project.bill_to_customer.name
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     32:             %small.text-muted (#{@project.bill_to_customer.matchcode})
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     33:
[e46b1d11-2144-4bc5-baa2-630d32338dfb]     34:         .row.mb-2
[e46b1d11-2144-4bc5-baa2-630d32338dfb]
[e46b1d11-2144-4bc5-baa2-630d32338dfb] app/views/projects/show.html.haml:31
[e46b1d11-2144-4bc5-baa2-630d32338dfb]
Information for cause: NoMethodError (undefined method `name' for nil):
[e46b1d11-2144-4bc5-baa2-630d32338dfb]
[e46b1d11-2144-4bc5-baa2-630d32338dfb] app/views/projects/show.html.haml:31
```

Also when editing a project, the "No customer (reusable project)" item is not prefilled for projects which have no customer assigned.

--- a/app/views/projects/show.html.haml
+++ b/app/views/projects/show.html.haml
@@ -28,8 +28,11 @@
           .col-sm-3
             %strong Bill to Customer:
           .col-sm-9
-            = @project.bill_to_customer.name
-            %small.text-muted (#{@project.bill_to_customer.matchcode})
+            - if @project.bill_to_customer
+              = @project.bill_to_customer.name
+              %small.text-muted (#{@project.bill_to_customer.matchcode})
+            - else
+              %span.text-muted No customer (reusable project)
 
         .row.mb-2
           .col-sm-3

--- a/test/controllers/projects_controller_test.rb
+++ b/test/controllers/projects_controller_test.rb
@@ -209,4 +209,52 @@ class ProjectsControllerTest < ActionDispatch::IntegrationTest
     assert_select 'input[type="checkbox"][name="project[active]"]'
     assert_select 'label.form-check-label', text: 'Active'
   end
+
+  test "should show project without customer" do
+    # Create a project without a customer
+    project_without_customer = Project.create!(
+      matchcode: 'NO_CUSTOMER',
+      description: 'Project without customer',
+      bill_to_customer: nil,
+      active: true
+    )
+
+    get project_url(project_without_customer)
+    assert_response :success
+    assert_select 'span.text-muted', text: 'No customer (reusable project)'
+  end
+
+  test "should edit project without customer" do
+    # Create a project without a customer
+    project_without_customer = Project.create!(
+      matchcode: 'NO_CUSTOMER',
+      description: 'Project without customer',
+      bill_to_customer: nil,
+      active: true
+    )
+
+    get edit_project_url(project_without_customer)
+    assert_response :success
+    # Form should render without errors
+    assert_select 'select[name="project[bill_to_customer_id]"]'
+    assert_select 'option', text: 'No customer (reusable project)'
+  end
+
+  test "should update project to remove customer" do
+    # Start with a project that has a customer
+    assert_not_nil @project.bill_to_customer
+
+    # Update to remove the customer
+    patch project_url(@project), params: {
+      project: {
+        bill_to_customer_id: '', # Empty to remove customer
+        description: 'Updated to have no customer'
+      }
+    }
+    assert_redirected_to project_url(@project)
+
+    @project.reload
+    assert_nil @project.bill_to_customer
+    assert_equal 'Updated to have no customer', @project.description
+  end
 end

